### PR TITLE
[1LP][WIPTEST] Fixed artifactor shutdown again

### PR DESCRIPTION
**Pull request by psav**
Another tweak

### CODE REVIEW

File: fixtures/artifactor_plugin.py [hunk 1]
```diff
@@ -249,14 +249,13 @@ def pytest_unconfigure(config):
 def shutdown(config):
     with lock:
         proc = config._art_proc
-        if proc:
+        if proc.returncode is None:
```
Comment (RonnyPfannschmidt) [inline fixtures/artifactor_plugin.py]: this holding true implies the code below is not correctly indented,

note that _art_proc can be None

Comment (psav) [inline fixtures/artifactor_plugin.py]: ```        if proc is not None and proc.returncode is None:``` better?

Comment (RonnyPfannschmidt) [inline fixtures/artifactor_plugin.py]: i'd use `getattr(proc, 'returncode', None) is None`

the logic below is also missing something, 
the `proc.wait` that is nested in here only makes sense when the process is not yet terminated and needs to be told

Comment (psav) [inline fixtures/artifactor_plugin.py]: right, the terminate above should initialize the termination

Comment (RonnyPfannschmidt) [inline fixtures/artifactor_plugin.py]: my mistake, the logic is not collapsible the way i wrote

**Comment by dajoRH** [comment]
Lint report for commit cedb45ff9361d701289c83a045e464281f759583:

`fixtures/merkyl.py`:
- :heavy_exclamation_mark: Line 97:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: fixtures/merkyl.py [hunk 2]
```diff
@@ -87,3 +87,11 @@ def test_test(merkyl_inspector):
                 print(merkyl_inspector.get_log('/path/to/log/file'))
     """
     return MerkylInspector(request)
+
+
+@pytest.yield_fixture(scope="session", autouse=True)
+def merkyl_setup(request):
+    fire_art_hook(request.config, 'setup_merkyl', ip=urlparse(env['base_url']).netloc)
```
Comment (RonnyPfannschmidt) [inline fixtures/merkyl.py]: this should use the appliance fixture, we may need to review appliance object initialization

File: fixtures/artifactor_plugin.py [hunk 3]
```diff
@@ -240,6 +239,7 @@ def pytest_runtest_logreport(report):
 @pytest.mark.hookwrapper
 def pytest_unconfigure(config):
     yield
+    fire_art_hook(config, 'finish_session')
```
Comment (RonnyPfannschmidt) [inline fixtures/artifactor_plugin.py]: this needs a check as it shouldn't run on worker nodes/slaves

**Comment by mshriver** [comment]
Marking WIPTEST, looks like there are a few unanswered questions. Came to review saving someone else the same click :)

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by jkrocil** [comment]
@psav @RonnyPfannschmidt is this still relevant?

**Pull request closed without merging**